libs/TouchSensorKit/source/TouchSensorKit.cpp

```diff
@@ -0,0 +1,208 @@
+// Leka - LekaOS
+// Copyright 2022 APF France handicap
+// SPDX-License-Identifier: Apache-2.0
+
+#include "TouchSensorKit.h"
+
+#include "rtos/ThisThread.h"
+
+using namespace leka;
+using namespace std::chrono_literals;
+
+void TouchSensorKit::init()
+{
+	for (Position position: _positions) {
+		initAtPosition(position);
+	}
+	_event_loop.registerCallback([this] { run(); });
+}
+
+void TouchSensorKit::start()
+{
+	stop();
+	_event_loop.start();
+}
+
+void TouchSensorKit::run()
+{
+	while (true) {
+		for (Position position: _positions) {
+			auto previousState = getLastStateAtPosition(position);
+			updateStateAtPosition(position);
+			auto currentState = getLastStateAtPosition(position);
+
+			if (!previousState && currentState && _on_sensor_touched_callback != nullptr) {
+				_on_sensor_touched_callback(position);
+			}
+			if (previousState && !currentState && _on_sensor_released_callback != nullptr) {
+				_on_sensor_released_callback(position);
+			}
+		}
+		rtos::ThisThread::sleep_for(100ms);
+	}
+}
+void TouchSensorKit::stop()
+{
+	_event_loop.stop();
+}
+
+void TouchSensorKit::calibrate(Position position)
+{
+	auto value = uint16_t {default_min_sensitivity_value};
+	auto step  = uint16_t {10};
+
+	constexpr auto accurate_read_count = uint8_t {15};
+	auto read_count					   = uint8_t {0};
+
+	setSensitivityAtPosition(position, value);
+
+	while (read_count < accurate_read_count && value < default_max_sensitivity_value) {
+		updateStateAtPosition(position);
+		if (!getLastStateAtPosition(position)) {
+			read_count = 0;
+			value += step;
+			setSensitivityAtPosition(position, value);
+		} else {
+			read_count++;
+		}
+		rtos::ThisThread::sleep_for(100ms);
+	}
+	setSensitivityAtPosition(position, value, true);
+}
+
+void TouchSensorKit::registerOnSensorTouched(std::function<void(Position)> const &on_sensor_touched_callback)
+{
+	_on_sensor_touched_callback = on_sensor_touched_callback;
+}
+
+void TouchSensorKit::registerOnSensorReleased(std::function<void(Position)> const &on_sensor_released_callback)
+{
+	_on_sensor_released_callback = on_sensor_released_callback;
+}
+
+void TouchSensorKit::initAtPosition(Position position)
+{
+	switch (position) {
+		case Position::ear_left:
+			return _ear_left.init();
+			break;
+		case Position::ear_right:
+			return _ear_right.init();
+			break;
+		case Position::belt_left_back:
+			return _belt_left_back.init();
+			break;
+		case Position::belt_left_front:
+			return _belt_left_front.init();
+			break;
+		case Position::belt_right_back:
+			return _belt_right_back.init();
+			break;
+		case Position::belt_right_front:
+			return _belt_right_front.init();
+			break;
+	}
+}
```

Comment:
Pour l'`init` fait tout directement dans la fonction plutôt que de passer par l'intermédiaire. D'autant que c'est le genre de fonction qui ne sera appelé qu'une seule fois et que ça fait sens de les faire tous en même temps, à la suite.

```suggestion
void TouchSensorKit::init()
{
	_ear_left.init();
	_ear_right.init();
	_belt_left_back.init();
	_belt_left_front.init();
	_belt_right_back.init();
	_belt_right_front.init();
	
	_event_loop.registerCallback([this] { run(); });
}
```